api/pkg/apis/v1alpha1/managers/stage/stage-manager.go

```diff
@@ -828,7 +825,9 @@ func (s *StageManager) HandleActivationEvent(ctx context.Context, actData v1alph
 		return nil, v1alpha2.NewCOAError(nil, "no stage found", v1alpha2.BadRequest)
 	}
 	if stageSpec, ok := campaign.Stages[stage]; ok {
-		if activation.Status != nil && activation.Status.Stage != "" && activation.Status.NextStage != stage {
+		if activation.Status != nil && activation.Status.StageHistory != nil && len(activation.Status.StageHistory) != 0 &&
+			activation.Status.StageHistory[len(activation.Status.StageHistory)-1].Stage != "" &&
+			activation.Status.StageHistory[len(activation.Status.StageHistory)-1].NextStage != stage {
```

Comment:
I think there would be cases where NextStage is empty, especially when there's a single stage in a campaign. The campaign wouldn't be able to trigger because of this check. Please test the scenario with a single stage.

Reply:
I have tried hello-world campaign sample which only has one stage and here is the result. 
<img width="353" alt="image" src="https://github.com/user-attachments/assets/021b940c-4c49-40e4-857f-2062440654c2">